src/du_parameters/frequency.py

```diff
@@ -0,0 +1,608 @@
+            raise NotImplementedError(f"Unsupported type for division: {type(other)}") from e
+
+    def __repr__(self) -> str:
+        """Return the Frequency as a string for development purposes."""
+        return f"Frequency({self.to_eng_string()})"
+
+    def __str__(self) -> str:
+        """Return the Frequency as a string."""
+        return self.to_eng_string()
+
+
+class ARFCN:
+    """Represents an Absolute Radio Frequency Channel Number (ARFCN) used in 5G.
+
+    Provides arithmetic operations and conversions from frequencies.
+
+    Args:
+        channel (int): The input ARFCN instance.
+
+    Raises:
+        ValueError: If the ARFCN is not within the valid range for ARFCN.
+        NotImplementedError: If the other value is not an ARFCN or int.
+    """
+
+    MIN_VALUE = 0
+    MAX_VALUE = 3279165
+
+    def __init__(self, channel: int):
+        if not isinstance(channel, int):
+            raise NotImplementedError("Channel must be an integer.")
+        if channel < self.MIN_VALUE or channel > self.MAX_VALUE:  # type: ignore[operator]
+            raise ValueError(
+                f"ARFCN must be between {self.MIN_VALUE} "
+                f"and {self.MAX_VALUE}, got {channel} instead."
+            )
+        self._channel = channel
+
+    def __repr__(self) -> str:
+        """Return the ARFCN as a string for development purposes."""
+        return f"ARFCN(channel={self._channel})"
+
+    def __str__(self) -> str:
+        """Return the ARFCN as a string."""
+        return str(self._channel)
+
+    def __add__(self, other: Any) -> "ARFCN":
+        """Add another ARFCN or integer to this ARFCN.
+
+        Args:
+            other (Any): The value to add.
+
+        Returns:
+            ARFCN: A new ARFCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an ARFCN or int or Decimal.
+        """
+        if isinstance(other, ARFCN):
+            return ARFCN(self._channel + other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return ARFCN(round(self._channel + other))  # type: ignore[operator]
+        raise NotImplementedError(f"Unsupported type for addition: {type(other).__name__}")
+
+    def __eq__(self, other: Any) -> bool:
+        """Check if ARFCN instance and other are equal.
+
+        Args:
+            other (Any): The value to compare with.
+
+        Returns:
+            ARFCN: A new ARFCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an ARFCN or int or Decimal.
+        """
+        if isinstance(other, int | Decimal):
+            return other == self._channel
+        if isinstance(other, ARFCN):
+            return other._channel == self._channel
+        raise NotImplementedError(f"Unsupported type for equality: {type(other).__name__}")
+
+    @staticmethod
+    def from_frequency(frequency: Frequency) -> "ARFCN":
```

Comment:
```suggestion
    @staticmethod
    def from_frequency(frequency: Frequency) -> "ARFCN":
```
```suggestion
    @classmethod
    def from_frequency(cls, frequency: Frequency) -> "ARFCN":
```
This should be a classmethod.